core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -0,0 +1,159 @@
+import java.io.InputStreamReader;
+import javax.annotation.Nullable;
+
+/**
+ * SSH into a remote machine and execute a script on that machine.
+ * A user must specify username and keypair authentication credentials.
+ * Optionals include port and machine URL
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("SSHAction")
+@Description("Action to run a script on remote machine")
+public class SSHAction extends Action {
+  private final SSHActionConfig config;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SSHAction.class);
+
+  public SSHAction(SSHActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(final ActionContext context) throws Exception {
+    Connection connection = new Connection(config.host);
+    try {
+      connection.connect();
+
+      String password = Strings.isNullOrEmpty(config.password) ? null : config.password;
+      if (!connection.authenticateWithPublicKey(config.user, new File(config.privateKeyFile), password)) {
+        throw new IOException(String.format("Unable to establish SSH connection for %s@%s on port %d",
+                                            config.user, config.host, config.port));
+      }
+
+      LOG.info(String.format("Connection established with the host %s", config.host));
+
+      Session session = connection.openSession();
+      session.execCommand(config.cmd);
+
+      InputStream stdout = new StreamGobbler(session.getStdout());
+      BufferedReader outBuffer = new BufferedReader(new InputStreamReader(stdout));
+      InputStream stderr = new StreamGobbler(session.getStderr());
+      BufferedReader errBuffer = new BufferedReader(new InputStreamReader(stderr));
+
+      StringBuilder outBuilder = new StringBuilder();
+      StringBuilder errBuilder = new StringBuilder();
+
+      String line = outBuffer.readLine();
+      while (line != null) {
+        outBuilder.append(line).append("\n");
+        line = outBuffer.readLine();
+      }
+
+      line = errBuffer.readLine();
+      while (line != null) {
+        errBuilder.append(line).append("\n");
+        line = errBuffer.readLine();
+      }
+
+      LOG.info("Output: {}", outBuilder.toString());
+      LOG.info("Errors: {}", errBuilder.toString());
+
+      if (config.output != null) {
+        context.getArguments().set(config.output, outBuilder.toString());
+      }
+      session.close();
+    } finally {
+      connection.close();
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    this.config.validate();
+  }
+
+  /**
+   * Config class that contains all the properties needed to SSH into the remote machine and run the script.
+   */
+  public static class SSHActionConfig extends PluginConfig {
+
+    @Description("Host name of the remote machine where the command need to be executed.")
+    private String host;
+
+    @Nullable
+    @Description("Port to connect to. Defaults to 22")
+    private int port;
```

Comment:
in order to set the default value, you should create a no-argument constructor and set the default value there.